includes/class-newspack-popups-segmentation.php

```diff
@@ -307,58 +328,58 @@ function( $item ) {
-			$wpdb->query( "ALTER TABLE {$events_table_name} CHANGE `created_at` `created_at` DATETIME NOT NULL" ); // phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.SchemaChange, WordPress.DB.DirectDatabaseQuery.NoCaching,WordPress.DB.PreparedSQL.InterpolatedNotPrepared
-		}
+	public static function check_update_version() {
+		$current_version = get_option( self::TABLE_VERSION_OPTION, false );
 
-		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
-		if ( $wpdb->get_var( $wpdb->prepare( 'SHOW TABLES LIKE %s', $transients_table_name ) ) != $transients_table_name ) {
-			$charset_collate = $wpdb->get_charset_collate();
-
-			$sql = "CREATE TABLE $transients_table_name (
-				option_id bigint(20) unsigned NOT NULL auto_increment,
-				option_name varchar(191) NOT NULL default '',
-				option_value longtext NOT NULL,
-				date datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
-				-- Date of the last update.
-				PRIMARY KEY  (option_id),
-				UNIQUE KEY option_name (option_name)
-			) $charset_collate;";
-
-			require_once ABSPATH . 'wp-admin/includes/upgrade.php';
-			dbDelta( $sql ); // phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.dbDelta_dbdelta
-			$wpdb->query( $wpdb->prepare( "INSERT INTO `{$transients_table_name}` (option_name, option_value) SELECT option_name, option_value FROM `{$wpdb->options}` WHERE option_name LIKE %s", "_transient%-popup%" ) ); // phpcs:ignore
+		if ( self::TABLE_VERSION !== $current_version ) {
+			self::create_database_tables();
+			update_option( self::TABLE_VERSION_OPTION, self::TABLE_VERSION );
 		}
 	}
 
+	/**
+	 * Create the clients and events tables.
+	 */
+	public static function create_database_tables() {
+		global $wpdb;
+		$reader_events_table_name = Segmentation::get_reader_events_table_name();
+		$readers_table_name       = Segmentation::get_readers_table_name();
+		$charset_collate          = $wpdb->get_charset_collate();
+
+		require_once ABSPATH . 'wp-admin/includes/upgrade.php';
+
+		$readers_sql = "CREATE TABLE $readers_table_name (
+			client_id varchar(100) NOT NULL,
+			date_created datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+			date_modified datetime NOT NULL,
+			reader_data longtext,
+			is_preview bool,
+			PRIMARY KEY  client_id (client_id),
+			KEY is_preview (is_preview),
+			KEY date_modified (date_modified)
+		) $charset_collate;";
+
+		dbDelta( $readers_sql ); // phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.dbDelta_dbdelta
+
+		$reader_events_sql = "CREATE TABLE $reader_events_table_name (
+			id bigint(20) NOT NULL AUTO_INCREMENT,
+			client_id varchar(100) NOT NULL,
+			date_created datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+			type varchar(20) NOT NULL,
+			context varchar(20) NOT NULL,
+			value longtext,
+			PRIMARY KEY  id (id),
+			KEY client_id_type_context (client_id, type, context),
+			KEY date_created (date_created)
```

Comment:
`client_id` index is not required with the composite `client_id, type, context`

Reply:
I just now realized that the index is created automatically due to the `NOT NULL` rule.